core/src/main/java/com/linecorp/armeria/common/HttpMessage.java

```diff
@@ -85,4 +86,28 @@
      */
     <T> StreamMessage<T> decode(HttpDecoder<T> decoder, ByteBufAllocator alloc,
                                 Function<? super HttpData, ? extends ByteBuf> byteBufConverter);
+
+    /**
+     * Transforms the {@link HttpData}s emitted by this {@link HttpMessage} by applying the
+     * specified {@link Function}.
+     */
+    HttpMessage mapData(Function<? super HttpData, ? extends HttpData> function);
+
+    /**
+     * Transforms the {@linkplain HttpHeaders trailers} emitted by this {@link HttpMessage} by applying the
+     * specified {@link Function}.
+     */
+    HttpMessage mapTrailers(Function<? super HttpHeaders, ? extends HttpHeaders> function);
+
+    /**
+     * Applies the specified {@link Consumer} to the {@link HttpData}s
+     * emitted by this {@link HttpMessage}.
+     */
+    HttpMessage peekData(Consumer<? super HttpData> action);
+
+    /**
+     * Applies the specified {@link Consumer} to the {@linkplain HttpHeaders trailers}
+     * emitted by this {@link HttpMessage}.
+     */
+    HttpMessage peekTrailers(Consumer<? super HttpHeaders> action);
```

Comment:
> How about moving up some common methods to HttpMessage such as peekData(), mapData(), mapError() and so on.

Based on this [comment](https://github.com/line/armeria/pull/3988#discussion_r775347408), I move up `mapData`, `mapTrailers`, `peekData`, `peekTrailers` to `HttpMessage`.

Reply:
It looks even better now. Thanks @injae-kim for the refactoring.